### PR TITLE
Adds Documention for how to run an in-place Agones upgrade

### PR DESCRIPTION
**Pull request by igooch**
**What type of PR is this?**

/kind documentation

**What this PR does / Why we need it**:

User documentation for how to perform an Agones in-place upgrade.

**Which issue(s) this PR fixes**:

Working on #3766

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/docs/Installation/upgrading.md
Comment (markmandel) [inline site/content/en/docs/Installation/upgrading.md:102]:
```suggestion
    3. This will start an update on the Fleet, wherein Ready `GameServers` will shut down and be recreated on
    the new configuration. Allocated `GameServers` need to wait until they are shut down and be
    replaced for to be updated with the new version of Agones' components.
```

I'd remove the reference to the `GameServerSet` since it's an implementation details the end user doesn't need to know/care about.

File: site/content/en/docs/Installation/upgrading.md
Comment (markmandel) [inline site/content/en/docs/Installation/upgrading.md:103]:
```suggestion
4.Wait until all the Ready `GameServers` are replaced.
```

Original statement is untrue, and also the end user rarely looks at GameServerSets, 

If there are allocated `GameServer` you could get multiple `GameServerSets`.

File: site/content/en/docs/Installation/upgrading.md
Comment (markmandel) [inline site/content/en/docs/Installation/upgrading.md:107]: Why is one recommended over the other?

File: site/content/en/docs/Installation/upgrading.md
Comment (markmandel) [inline site/content/en/docs/Installation/upgrading.md:109]: Rather than have this section, could we not have 

#### In-Place Agones Upgrades

... outline how to do it in in place

##### Upgrade to Existing Fleet (recommended)

... outline how to force a `Fleet` update

and then we don't have to have the same thing twice?

File: site/content/en/docs/Installation/upgrading.md
Comment (igooch) [reply site/content/en/docs/Installation/upgrading.md:83]: Either one will kick off a Fleet update. I'm suggesting the labels here, so that they can use `matchLabels` to create a gameserverallocation like:

```
apiVersion: allocation.agones.dev/v1
kind: GameServerAllocation
spec:
  selectors:
    - matchLabels:
        release: 1.42.0
```

File: site/content/en/docs/Installation/upgrading.md
Comment (igooch) [reply site/content/en/docs/Installation/upgrading.md:107]: Do you mean compared to delaying the upgrade, or compared to adding or not adding a label?

File: site/content/en/docs/Installation/upgrading.md
Comment (igooch) [reply site/content/en/docs/Installation/upgrading.md:103]: Allocated game servers will still be at the old configuraiton until they are set back to ready, shut down, etc. So all the Ready game servers could be at the new configuration, and there would still be game servers at the old configuraiton. I guess, what are we defining as "the rollout is complete"? It could be either that Agones is upgraded (this should be in the ~minutes), or all Ready game servers are at the new configuraiton, or all game servers are at the new configuraiton?

Comment (igooch) [reply site/content/en/docs/Installation/upgrading.md:104]: In regards to when a _Fleet_ rollout is complete:

```
"Done" (for the initial Alpha release) is:

At least one Ready or Allocated or Reserved GameServer from the new GameServerSet.
Only Allocated and Reserved GameServers on any previous GameServerSets.
```
https://github.com/googleforgames/agones/issues/2817#issuecomment-1526003786

Should this apply to the Agones upgrade as well?

File: site/content/en/docs/Installation/upgrading.md
Comment (markmandel) [reply site/content/en/docs/Installation/upgrading.md:59]: Easy option - make it 1.99.0 until we're happy with where it is at, and we can pull it back to whatever the next release is? 😄

File: site/content/en/docs/Installation/upgrading.md
Comment (markmandel) [reply site/content/en/docs/Installation/upgrading.md:84]: Nit on rewrite:

```markdown
If you have no changes to make to the existing `spec.template` in the fleet.yaml, then either add a label or annotation to the `spec.template.metadata`. Reapply with `kubectl apply -f fleet.yaml`.
```

(also being consistent on how we're referencing each part of the CRD)

I don't know if we need to be providing recommendation on label vs annotation here -- just because it's entirely up to the end user -- they may not care about filtering on it on allocation and it is a simpler instruction without it.

WDYT?

File: site/content/en/docs/Installation/upgrading.md
Comment (markmandel) [reply site/content/en/docs/Installation/upgrading.md:108]: Compared to delaying the existing Fleet - we never specify why one is preferred over the other. We should be clear on that (I'm also curious 😆)

File: site/content/en/docs/Installation/upgrading.md
Comment (markmandel) [reply site/content/en/docs/Installation/upgrading.md:104]: I'd be tempted to be a little more ambiguous here. Probably the best we could do would be "Wait until all the previous Ready GameServers have been replaced by the new version.'

And leave it up to the end user what they want to do with anything Allocated or Reserved.

File: site/content/en/docs/Installation/upgrading.md
Comment (igooch) [reply site/content/en/docs/Installation/upgrading.md]: I did some rearranging / reordering. The wording or format could probably be clearer. Let me know if you have any additional suggestions!

File: site/content/en/docs/Installation/upgrading.md
Comment (gongmax) [inline site/content/en/docs/Installation/upgrading.md:108]: liklihood -> likelihood

File: site/content/en/docs/Installation/upgrading.md
Comment (igooch) [inline site/content/en/docs/Installation/upgrading.md:55]: Do we want to publish this now, or wait for the release? There's no feature gate, so there's nothing preventing a user from starting to use in place upgrades now.

Comment (gongmax) [reply site/content/en/docs/Installation/upgrading.md:55]: I think it's still make sense to have a defined time point when the feature is "officially" in alpha. So let's keep this.

**Review by gongmax — APPROVED** [review]

**Pull request merged**
